openwisp_controller/api/urls.py

```diff
@@ -0,0 +1,40 @@
+from django.urls import include, path
+
+from . import views as api_views
+
+urlpatterns = [
+    path(
+        'controller/',
+        include(
+            [
+                path('template/', api_views.template_list, name='template_list',),
+                path(
+                    'template/<str:pk>/',
+                    api_views.template_detail,
+                    name='template_detail',
+                ),
+                path(
+                    'template/<str:pk>/configuration/',
+                    api_views.download_template_config,
+                    name='download_template_config',
+                ),
+                path('vpn/', api_views.vpn_list, name='vpn_list',),
+                path('vpn/<str:pk>/', api_views.vpn_detail, name='vpn_detail',),
+                path(
+                    'vpn/<str:pk>/configuration/',
+                    api_views.download_vpn_config,
+                    name='download_vpn_config',
+                ),
+                path('device/', api_views.device_list, name='device_list',),
+                path(
+                    'device/<str:pk>/', api_views.device_detail, name='device_detail',
```

Comment:
Quick Q: shouldn't the details pages be read-only and we should have separate edit pages. 

Currently, I went to `http://127.0.0.1:8000/api/v1/controller/device/<pk>/`

and it gave me details and the form to edit the device! :smile:

Reply:
Yes, @atb00ker I had a discussion about this with Federico, to have separate endpoints for GET/PATCH/PUT of the device detail and other endpoints but he suggested me to keep it in this way. :smile:

Reply:
Thanks for the info, I found the comment: https://github.com/openwisp/openwisp-controller/issues/379#issuecomment-785244645

Reply:
https://www.amazon.com/RESTful-Web-Services-Leonard-Richardson/dp/0596529260

Reply:
Thanks for the recommendation, I'll check it out! :smile: